### PR TITLE
chore: upgrade prettier to 3.2.5 and prettier-plugin-solidity to 1.3.1

### PR DESCRIPTION
**Pull request by yonadaa**
Pulled out from https://github.com/latticexyz/mud/pull/2302 (Solidity prettier needs to support events at file level)

### CODE REVIEW

**Comment by yonadaaa** [comment]
TIL we can remove `@types/prettier`: https://www.npmjs.com/package/@types/prettier

**Review by holic — DISMISSED** [review]
lgtm! may need to be strategic about when we land this since this'll prob add merge conflicts to ~all PRs

File: docs/pages/store/reference/misc.mdx [hunk 1]
```diff
@@ -269,6 +269,62 @@ function slice32(bytes memory data, uint256 start) internal pure returns (bytes3
 | -------- | --------- | -------------------------------------------------------------------------- |
 | `output` | `bytes32` | The extracted bytes32 value from the specified position in the bytes blob. |
 
+## FieldLayout_Empty
```
Comment (holic) [inline docs/pages/store/reference/misc.mdx:272]: I feel like I saw this in my PR too and IIRC had to clear out all the various foundry caches (cache, out, docs) before rebuilding/regenerating docs. Try that and see if this disappears?

Comment (yonadaaa) [reply docs/pages/store/reference/misc.mdx:272]: Good idea, I've been playing with it to no results

Comment (holic) [reply docs/pages/store/reference/misc.mdx:272]: also be sure this has latest main too

**Pull request merged**